configs/ci/build-only.yaml

```diff
@@ -0,0 +1,5 @@
+plugin_group:
+  - !include "plugins/debugview.yaml"
+  - !include "plugins/gldemo.yaml"
+  - !include "plugins/timewarp_gl.yaml"
+  - !include "plugins/pose_lookup.yaml"
```

Comment:
These three will get compiled and run as part of the CI flows. Shouldn't `build-only.yaml` only be for plugins that can't be run at all in CI?

Reply:
The interface for specifying plugins (across this entire PR refactor) is designed to be verbose, but explicit. This makes it clearer how things connect. The file structure is designed so that adding and removing new plugins, or editing properties about specific data/plugins generates the fewest conflicts for existing configs/plugins

In terms of dependencies, again, the interface is explicit here. We indicate everything that should be built. The CI doesn't assume anything about the dependence structure, or properties of the application that might _change_ over time. Instead, it allows for a smarter implementation within Runner or a new build system to detect the dependencies and alter the compilation strategy accordingly.

Following this idea, the verbosity of the plumbing can be reduced later by augmenting Runner to do more automatic detection of plugins.

Reply:
> In terms of dependencies, again, the interface is explicit here. We indicate everything that should be built. The CI doesn't assume anything about the dependence structure, or properties of the application that might change over time. Instead, it allows for a smarter implementation within Runner or a new build system to detect the dependencies and alter the compilation strategy accordingly.

I add a note here that this design is also motivated by maintenance overhead. If we handle the strategies in the Runner/scripts, its much easier for an ILLIXR dev, new user, or plugin writer to understand what the plumbing changes are (similar to reducing conflicts above).